### ISSUE TITLE
Create a schematic for every console controller currently used in RetroPie Libretro cores

### ISSUE THREAD

**Issue by taalas** [taalas/RetroPie-Artwork#1]
We currently emulate the following console systems:

3do
Atari 2600
Atari 5200
Atari 7800
Atari Jaguar
Atari Lynx
Bandai Wonderswan (Color)
Dreamcast
Genesis/Megadrive
Game Gear
Game Boy
Game Boy Color
Game Boy Advance
Intellivision
MasterSystem
Nintendo 64
Nintendo DS
Nintendo Entertainment System
Neo Geo
Neo Geo Pocket
PC Engine/ TurboGrafx 16
PlayStation 1
Sega 32X
Sega CD
Sega Saturn
Sega SG-1000
Super Nintendo Entertainment System
Vectrex
Videopac or Odyssey2
Virtual Boy

**Comment by taalas** [comment]
Still open (and libretro core):

Bandai Wonderswan (Color)
PlayStation 1

**Comment by taalas** [comment]
I will remove the Sega 32x and Sega CD for now, since those are Mega Drive addons

**Comment by taalas** [comment]
Added Mattel Intellivision and Gameboy Advance files

**Comment by HerbFargus** [comment]
btw as far as I understand it, Sega Dreamcast, Intellivision, and Atari 5200 aren't part of retroarch. they look good though.

**Comment by taalas** [comment]
You are right. Don't know how I got confused there...oh well, perhaps we can somehow fit them in with the other emulator pages ;)

**Comment by HerbFargus** [comment]
I am sure we will find places for all of them. Florian was saying he'll eventually incorporate non-retroarch emulators into inputstation as well, so they will have a use in the future. Also one console you're forgetting on your list is PC Engine/ TurboGrafx 16 (which is a retroarch core).

**Comment by taalas** [comment]
I updated the lists above

**Comment by HerbFargus** [comment]
Wonderswan, Neo Geo Pocket, and Nintendo DS have just been added to RetroPie as well (All experimental at the moment of course)

**Comment by taalas** [comment]
Thanks, I followed the merges on the original repo but forgot to add them here...will try to finish them up soon hopefully. Currently trying to deal with some unforeseen problems -.-

**Comment by HerbFargus** [comment]
ha ha its always something- no worries, I'm otherwise preoccupied for the next few weeks but once I have some free time I'll work on the rest of the diagrams- what you have looks good so far :)

**Comment by taalas** [comment]
yep, I am sorry for the delay :(

**Comment by taalas** [comment]
added the Videopac, Vectrex and SG-1000 controllers

**Comment by HerbFargus** [comment]
So I guess some later versions of the genesis 6 button controllers had a mode button that in retroarches case would function as the select button/hotkey
http://blog.petrockblock.com/forums/topic/getting-the-mode-button-to-work-on-a-genesis-6-button/

Perhaps a slight tweak to add it may be useful

http://nintendoagemedia.com/users/7347/photobucket/1B04A99C-DFB4-ECEC-850620F7A4FD0AAF.jpg

Sorry I've been slacking on the diagrams, I just got back this weekend so I'll try and get them all done this week.

**Comment by taalas** [comment]
No worries, I still have to to do some of the controllers. Will finish them this week hopefully.

I will have a look at the alternative 6 button genesis pad and add it to our database, great find!

**Comment by HerbFargus** [comment]
So I'm finding myself perplexed on how to approach these last few controllers. The two most perplexing are the Atari Jaguar (which is pretty much unplayable on a pi) and the Intellivision. I know that free5ty1e has been doing a lot of work in getting the dreamcast emulator to be configurable for controllers and once PetRockBlog finishes these last tests for getting inputstation stable he'll probably start incorporating non-retroarch emulators into the code such as reicast and the intellivision. The atari Jaguar varies greatly between games as it was originally meant to have overlays that you would place on the controller for each game. What do you think would be the best approach for the Atari Jaguar, Dreamcast, and Intellivision?

**Comment by taalas** [comment]
I agree that those controllers are a little problematic for now and would suggest the following:

Dreamcast: Once we have a final mapping function for Inputstation -> Reicast this controller should be fine for mapping. For now we wait til Reicast integration and Inputstation is finalized.

Intellivision: Since it is not a Libretro core we might be better off just dropping it for now. I remember playing an Intellivision back in the day and the cards that modified the pad. Due to the number of buttons on this controller the emulator maps this to keys on the keyboard. Not sure if playing Intellivision on a recent controller makes sense at all, what do you think?

Jaguar: Haven't tried the emulation on the RetroPie myself yet tbh. I didn't know that these pads had overlays as well. We could do a mapping for the "standard" controller buttons and leave the variable pad for now. The Libretro wiki page for the jaguar core does not show any mappings, so I am not sure what the standard mappings from RetroPad are, do you know?

Perhaps we should postpone these pads for now and come back to them once integration is a little clearer?

**Comment by HerbFargus** [comment]
That sounds like a good idea to me. For the jaguar I was just going to test random games until I figured out all the buttons, but at the very least I could probably do the main 3 buttons. For the other ones I'll probably just insert a picture of the blank controller on the wiki until the controls are incorporated into inputstation. Thanks

**Comment by taalas** [comment]
I will submit another batch of controllers/consoles soon. Hopefully will cover most of what is left. I will contact you privately sometime soon to brainstorm some other ideas as well!

**Comment by HerbFargus** [comment]
Latest 6 button rendition looks good. I was looking at the original 6 button model and all the places that I looked The start button was in the middle rather than above the buttons. The original 3 button genesis controller had it diagonally above the buttons like you showed but unless there was a separate controller for the megadrive I think the start button was in the middle for the 6 button controller.

http://segaretro.org/Control_Pad_(Mega_Drive)

http://segaretro.org/Six_Button_Control_Pad_(Mega_Drive)

**Comment by taalas** [comment]
I think I accidentally used the 6 button wireless controller as a reference for the first version, which is also an official Sega product but not very widely used:

http://www.consolepassion.co.uk/products/sega-megadrive/hardware/sega-megadrive-6-button-wireless-controllers-loose

I will rename the file accordingly. Great find! Thanks for letting me know! I didn't really remember from personal experience since all I ever used back in the day was the 3 button version...

**Comment by HerbFargus** [comment]
Ah I had completely forgotten they had a wireless controller (Sega had a lot of peripherals...)  Its fine as it is then, no sense in creating 8 different variation of the same controller just due to button placement :p

**Comment by taalas** [comment]
Yeah, it was my bad...I didn't realize there were so many before it was too late ;)
